examples/c/ex_col_store.c

```diff
@@ -0,0 +1,519 @@
+    if (exact == -1)
+        ret = start_time_cursor->next(start_time_cursor);
+
+    if (ret == 0)
+        error_check(session->join(session, join_cursor, start_time_cursor, "compare=ge"));
+    else
+        return ret;
+
+    /* Find the ending record closest to desired end time. */
+    end_time_cursor->set_key(end_time_cursor, end_time);
+    error_check(end_time_cursor->search_near(end_time_cursor, &exact));
+    if (exact == 1)
+        ret = end_time_cursor->prev(end_time_cursor);
+
+    if (ret == 0)
+        error_check(session->join(session, join_cursor, end_time_cursor, "compare=le"));
+    else
+        return ret;
+
+    /* Initialize minimum temperature and maximum temperature to temperature of the first record. */
+    ret = join_cursor->next(join_cursor);
+    if (ret != 0)
+        return ret;
+
+    error_check(join_cursor->get_key(join_cursor, &recno));
+    error_check(join_cursor->get_value(join_cursor, &hour, &temp));
+    *min_temp = temp;
+    *max_temp = temp;
+
+    /* Iterating through found records between start and end time to find the min & max temps. */
+    while ((ret = join_cursor->next(join_cursor)) == 0) {
+        error_check(join_cursor->get_key(join_cursor, &recno));
+        error_check(join_cursor->get_value(join_cursor, &hour, &temp));
+
+        if (temp < *min_temp)
+            *min_temp = temp;
+
+        if (temp > *max_temp)
+            *max_temp = temp;
+
+        /* For debugging purposes */
+        /*
+            printf("ID %" PRIu64, recno);
+            printf(": hour %" PRIu16 " temp: %" PRIu8 "\n", hour, temp);
+         */
+    }
+
+    return (EXIT_SUCCESS);
+}
+
+/* Obtains the average data across all fields given a specific location. */
+void
+average_data(WT_SESSION *session)
+{
+    WT_CURSOR *loc_cursor;
+    uint64_t recno;
+    unsigned int count;
+    uint16_t hour, loc_lat, loc_long, pressure;
+    uint8_t feels_like_temp, humidity, temp, wind;
+    const char *country, *day;
+
+    /* num_rec is the total number of records we're obtaining averages for. */
+    int ret, num_rec;
+    /* rec_arr holds the sum of the records in order to obtain the averages. */
+    unsigned int rec_arr[5] = {0, 0, 0, 0, 0};
+
+    /* Open a cursor to search for the location, currently RUS. */
+    error_check(session->open_cursor(session, "index:weather:country", NULL, NULL, &loc_cursor));
+    loc_cursor->set_key(loc_cursor, "RUS\0\0");
+    ret = loc_cursor->search(loc_cursor);
+
+    /*
+     *  Error handling in the case RUS is not found. In this case as it's a hardcoded location,
+     *  if there aren't any matching locations, no average data is obtained and we proceed with the
+     *  test instead of aborting. If an unexpected error occurs, exit.
+     * */
+    if (ret == WT_NOTFOUND)
+        return;
+    else if (ret != 0)
+        exit(1);
+
+    /* Populate the array with the totals of each of the columns. */
+    count = 0;
+    while ((ret = loc_cursor->next(loc_cursor)) == 0) {
```

Comment:
This doesn't look correct.

When search is successful, the cursor is on a record with location as "RUS". Doing a next() makes the cursor go to the next record, which means you ignore the first record with location "RUS". Also, then the while loop continues to the end of the database, and would include records that are not from "RUS", so you need to skip those:
```
if (country != RUS)
    continue;
```